### PR TITLE
!WIP: 470

### PR DESCRIPTION
**Pull request by nathanlcarlson**
This is a work in progress and is not ready for merging at this time. Let me know your thoughts on this type of object oriented config management. It is regarding #470 .

### CODE REVIEW

File: esg_init.py [hunk 1]
```diff
@@ -145,14 +193,30 @@ def __init__(self):
...
+        self.truststore_file = os.path.join(self.tomcat_conf_dir, "esg-truststore.ts")
+        self.truststore_password = "changeit"
     def init_directories(self):
         directories = [
             self.tomcat_conf_dir
         ]
         for directory in directories:
             pybash.mkdir_p(directory)
 
+class CDATConfig(BaseConfig):
```
Comment (William-Hill) [inline esg_init.py]: This could be removed as CDAT is not setup during the bootstrap

File: esg_init.py [hunk 2]
```diff
@@ -334,77 +330,26 @@ def init():
...
-    postgress_driver = "org.postgresql.Driver"
-    postgress_protocol = "jdbc:postgresql:"
-    pg_sys_acct = "postgres"
-    pg_sys_acct_group = pg_sys_acct
-    publisher_repo = "git://github.com/ESGF/esg-publisher.git"
-    apache_frontend_repo = "https://github.com/ESGF/apache-frontend.git"
-    publisher_repo_https = "https://github.com/ESGF/esg-publisher.git"
-    esgcet_egg_file = "esgcet-%s-py%s.egg" % (esgcet_version, python_version)
+
     esg_testdir = workdir + "/../esg_test"
```
Comment (William-Hill) [inline esg_init.py]: This can be removed

File: esg_init.py [hunk 3]
```diff
@@ -238,84 +286,32 @@ def init():
...
-        os.environ["ESGINI"]
-    except KeyError:
-        publisher_home = os.path.join(esg_config_dir, "esgcet")
-        publisher_config = "esg.ini"
-        ESGINI = os.path.join(publisher_home, publisher_config)
 
     ############################################
     ####  DO NOT EDIT BELOW THIS POINT!!!!! ####
     ############################################
     os.environ["GIT_SSL_NO_VERIFY"] = "1"
```
Comment (William-Hill) [inline esg_init.py]: Theses os.environ can be deprecated. Or they need to be refactored to be written out to a separate file

**Comment by William-Hill** [comment]
I like the concept.  I'll want to do a test installation with this branch to see if it all works.  If all goes well, I'll merge it in

**Comment by nathanlcarlson** [comment]
This isn't ready for merging/testing yet. Still need to do the imports in the files where the config is needed and finish setting up the classes.

**Pull request closed without merging**